password-instances/src/Data/Password/Instances.hs

```diff
@@ -14,90 +14,79 @@ Maintainer  : cdep.illabout@gmail.com
 import Web.HttpApiData (FromHttpApiData(..))
 
 
 -- $setup
 -- >>> :set -XOverloadedStrings
+-- >>> :set -XDataKinds
 --
 -- Import needed functions.
 --
 -- >>> import Data.Aeson (decode)
+-- >>> import Data.Password (Salt(..), unsafeShowPassword)
+-- >>> import Data.Password.Scrypt (defaultParams, hashPasswordWithSalt)
 -- >>> import Database.Persist.Class (PersistField(toPersistValue))
 -- >>> import Web.HttpApiData (parseUrlPiece)
 
--- | This instance allows a 'Pass' to be created from a JSON blob.
+-- | This instance allows a 'Password' to be created from a JSON blob.
 --
--- >>> let maybePass = decode "\"foobar\"" :: Maybe Pass
--- >>> fmap unsafeShowPassword maybePass
+-- >>> let maybePassword = decode "\"foobar\"" :: Maybe Password
+-- >>> fmap unsafeShowPassword maybePassword
 -- Just "foobar"
 --
--- There is no instance for 'ToJSON' for 'Pass' because we don't want to
--- accidentally encode a plain-text 'Pass' to JSON and send it to the end-user.
+-- There is no instance for 'ToJSON' for 'Password' because we don't want to
+-- accidentally encode a plain-text 'Password' to JSON and send it to the end-user.
 --
--- Similarly, there is no 'ToJSON' and 'FromJSON' instance for 'PassHash'
+-- Similarly, there is no 'ToJSON' and 'FromJSON' instance for 'PasswordHash'
 -- because we don't want to accidentally send the password hash to the end
 -- user.
-instance FromJSON Pass where
-  parseJSON = fmap mkPass . parseJSON
+instance FromJSON Password where
+  parseJSON = fmap mkPassword . parseJSON
 
--- | This instance allows a 'Pass' to be created with functions like
+-- | This instance allows a 'Password' to be created with functions like
 -- 'Web.HttpApiData.parseUrlPiece' or 'Web.HttpApiData.parseQueryParam'.
 --
--- >>> let eitherPass = parseUrlPiece "foobar"
--- >>> fmap unsafeShowPassword eitherPass
+-- >>> let eitherPassword = parseUrlPiece "foobar"
+-- >>> fmap unsafeShowPassword eitherPassword
 -- Right "foobar"
-instance FromHttpApiData Pass where
-  parseUrlPiece = fmap mkPass . parseUrlPiece
+instance FromHttpApiData Password where
+  parseUrlPiece = fmap mkPassword . parseUrlPiece
 
--- | This instance allows a 'PassHash' to be stored as a field in a database using
+-- | This instance allows a 'PasswordHash' to be stored as a field in a database using
 -- "Database.Persist".
 --
 -- >>> let salt = Salt "abcdefghijklmnopqrstuvwxyz012345"
--- >>> let pass = mkPass "foobar"
--- >>> let hashedPassword = hashPassWithSalt salt pass
+-- >>> let pass = mkPassword "foobar"
+-- >>> let hashedPassword = hashPasswordWithSalt defaultParams salt pass
 -- >>> toPersistValue hashedPassword
--- PersistText "14|8|1|YWJjZGVmZ2hpamtsbW5vcHFyc3R1dnd4eXowMTIzNDU=|nENDaqWBmPKapAqQ3//H0iBImweGjoTqn5SvBS8Mc9FPFbzq6w65maYPZaO+SPamVZRXQjARQ8Y+5rhuDhjIhw=="
+-- PersistText "16|8|1|YWJjZGVmZ2hpamtsbW5vcHFyc3R1dnd4eXowMTIzNDU=|BH0oidcU/4Ec7Co4EM+LQ6xp39//MnOUhqmNeOnOz/nl4JHNXEJBw5dPdi3wTStYr+e1SmJkzHJrMvUJYNxK1w=="
```

Comment:
Note to self: Add an item to the changelog that says the hash produced by `hashPasswordWithSalt` has changed in this new version.

Reply:
Well, `hashPasswordWithSalt` is completely different, since it takes parameters now. And the hash is only different because the rounds are higher, if you change `defaultParams` to `defaultParams{ scryptRounds = 14 }`, it'll produce the same hash.

Reply:
@Vlix I was mostly just worried about people that are currently using `password` in production.  They might want an easy way to produce the password hashes in the same format.

Of course, I haven't actually heard from anyone using `password` in production, so maybe it doesn't matter at all.

Reply:
The format is not changed. I've even got tests that verify it in `test/Scrypt.hs`

We can set the default cost back to 14 until we get benchmarks that provide a more sensible default value (like I was planning for `bcrypt` anyway) so that the hashes stay the same if using `defaultParams`?

Reply:
> The format is not changed.

Ah, I see, so it is just the cost that has been changed from 14 to 16.

Okay, how about this:

We leave it as you currently have it in the PR, but you add a small note to the `CHANGELOG` that tells people to set the `cost` argument in the bcrypt `defaultParams` if they want the same functionality from the old version of `password`?

Although, like I said above, I don't know if there is really anyone using `password`, so if you don't want to do this, I'd be fine with going ahead and doing a release anyway.

I don't want my small nitpicks to hold up a new release.

Reply:
Well, it's a fair point, and since I increased the rounds just because the python library has 16 as default, and it seemed fast enough in the test cases, I think it's better to just keep it at 14 and do all the default tweaking after the benchmarking is done.
This makes it more in line with the thought of the other defaults as well.

Reply:
Okay, that sounds good to me.

Reply:
Updated